### ISSUE TITLE
How to combine fret and cocosim to a fret-cocosim workflow?

### ISSUE THREAD

**Issue by marious123g** [NASA-SW-VnV/fret#35]
Hi, all. I've read the paper _Bridging the Gap Between Requirements and Model Analysis_ and want to run an example of fret-cocosim workflow but I dont know how to combine them. Like how can I use the exported zip file? Is there any interface on the front-end in cocosim to import? I would appreciate it if there is any tutorial.

**Comment by anmavrid** [comment]
Hello @marious123g,

Thanks for reading our paper. If you have already installed CoCoSim (installation instructions: https://github.com/NASA-SW-VnV/CoCoSim/blob/master/INSTALL.md), here is a video demo with the steps that you need to follow on the CoCoSim side: https://drive.google.com/file/d/1wcM4dYThECuhDD7I_-mi9c3mQdjHYzSv/view?usp=sharing (this was part of our conference demo).

I hope this is helpful.

Anastasia

**Comment by SoftPro** [comment]
Hello @anmavrid 
This is the cocosim I installed on MATLAB，![image](https://user-images.githubusercontent.com/89380949/145368204-ad75c91f-2f44-4a28-a952-fa26d6e60e9f.png)  。
This is cocosim in the video，
![image](https://user-images.githubusercontent.com/89380949/145368587-60bd69f0-9272-42a5-83bb-31f29533598d.png)。
I want to ask why there are no options after installation. Is it because of the version？

**Comment by andreaskatis** [comment]
Hello @SoftPro,

How did you install CoCoSim? I suggest using the latest release: https://github.com/NASA-SW-VnV/CoCoSim/releases/tag/v1.2

**Comment by SoftPro** [comment]
OK，thank you！



---Original---
From: "Andreas ***@***.***&gt;
Date: Thu, Dec 9, 2021 17:33 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [NASA-SW-VnV/fret] How to combine fret and cocosim to afret-cocosim workflow? (Issue #35)




 
Hello @SoftPro,
 
How did you install CoCoSim? I suggest using the latest release: https://github.com/NASA-SW-VnV/CoCoSim/releases/tag/v1.2
 
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by SoftPro** [comment]
Hello @andreaskatis
I installed cocosim according to the method you provided, but I don't know why there is no requirement in it, and then I can't import the requirements. How can I solve this problem?
![image](https://user-images.githubusercontent.com/89380949/145584339-2035f2a4-61b9-44a0-90b4-434206aa7ac1.png)

**Comment by andreaskatis** [comment]
Hi @SoftPro,

Unfortunately I do not currently have access to the tool to try and replicate your issue. Are you sure that you are launching CoCoSim from the directory where 'start_cocosim.m' exists for version v1.2?

If yes, can you try v1.1 and see if that works instead? https://github.com/NASA-SW-VnV/CoCoSim/releases/tag/v1.1

**Comment by SoftPro** [comment]
Hello @andreaskatis
When I use v1 1, the same problem occurred.However, they all reported the same error, as shown in the figure:
![image](https://user-images.githubusercontent.com/89380949/145660779-162309cd-70c5-496b-96ac-4131069ad660.png)
I'd like to ask if Ubuntu version needs 20.04, and I'm using 18.04, or any other questions?

**Comment by andreaskatis** [comment]
Hi @SoftPro,

Okay, so the installation did not complete successfully. I assume that the error is popping up some time after trying to install using the 'install_cocosim' script under 'CoCoSim/scripts'.

The error message mentions a log file 'bootstrap.log' under 'tools/build'. Can you please attach this file here?

**Comment by SoftPro** [comment]
***@***.***
&nbsp; &nbsp; &nbsp; I will bootstrap Log file has been added to the attachment. I hope you can help solve the following problems. It seems to be the version of OPAM.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "NASA-SW-VnV/fret"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年12月11日(星期六) 下午2:56
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [NASA-SW-VnV/fret] How to combine fret and cocosim to a fret-cocosim workflow? (Issue #35)





 
Hi @SoftPro,
 
Okay, so the installation did not complete successfully. I assume that the error is popping up some time after trying to install using the 'install_cocosim' script under 'CoCoSim/scripts'.
 
The error message mentions a log file 'bootstrap.log' under 'tools/build'. Can you please attach this file here?
 
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by andreaskatis** [comment]
Hi @SoftPro,

I don't see any attachment in your reply, but if it is an issue with the opam version, I would suggest switching to v2.0.5, which is the one I use and know for sure it currently works.

**Comment by SoftPro** [comment]
hello，@andreaskatis
Thank you very much for your help. I changed the OPAM version and solved the previous installation problem, but there is still no option to add requirements in Simulink. The following sentence appears after installation. I have also executed it, but it is still ineffective. May there be other problems?
![image](https://user-images.githubusercontent.com/89380949/145697805-932ac408-b222-44c2-afbf-0f1cb90c9132.png)

**Comment by andreaskatis** [comment]
Hi @SoftPro,

All I can guess at the moment is that you are are not starting CoCoSim properly. Please make sure that you follow the steps below :

1. You extract the release .zip file and run the install_cocosim script.
2. When (1) is completed, you have to open up MATLAB and navigate using MATLAB's explorer window to the home directory of the downloaded CoCoSim release. According to your last screenshot this means that you have to navigate into '/home/cs18/Desktop/CoCoSim_v1.2'. Then, start CoCoSim from the MATLAB terminal by typing 'start_cocosim'. This step is something that you have to do every time that you start a fresh instance of MATLAB.

Unfortunately, I cannot currently provide visual guidance in the form of screenshots, but the most important thing is to make sure that you have navigated into CoCoSim's home folder within the MATLAB file explorer, before starting CoCoSim. 

Can you confirm that you are performing the steps, but still see different options? If yes, did you also try using v1.1 after resolving the previous issues in this thread?

**Comment by SoftPro** [comment]
Hello，andreaskatis
You can refer to this picture. It seems that it is not the reason to start cocosim. I start cocosim V1 in this way 1 for the same reason, there is no option to import requirements.
![image](https://user-images.githubusercontent.com/89380949/145703876-903ffb54-1653-4def-81c5-a4b4722ed2ea.png)

**Comment by andreaskatis** [comment]
Looking at the screenshot, it appears that when you run 'start_cocosim', the configuration step skips the part that synchronizes with the necessary remote repositories.

This is probably happening because you are either running this on a machine without internet connection, or a connection that blocks direct access to github.

In this case, you would have to complete the configuration manually. This can be done by looking at what the 'scripts/install_cocosim_lib.m' does to fetch and install dependencies, and perform each one of them manually.

**Comment by SoftPro** [comment]
Hello,@andreaskatis
I'm sorry to bother you again, but there are still some questions I don't understand. Executing install_ At the time of cocosim, my lustrec has been successfully installed. It seems that the import requirements depend on this lustrec file. In the “cocosim / tools / filters / OSX / bin /”  directory, you can find binary files such as kind2 and lustrec, and I run start_ Why does start_cocosim skip repositories because of the matlab version? I use the 2021a version.

**Comment by SoftPro** [comment]
Hello,@andreaskatis
I'm sorry to bother you again, but there are still some questions I don't understand. Executing install_ At the time of cocosim, my lustrec has been successfully installed. It seems that the import requirements depend on this lustrec file. In the “cocosim / tools / filters / OSX / bin /”  directory, you can find binary files such as kind2 and lustrec, and I run start_ Why does start_cocosim skip repositories because of the matlab version? I use the 2021a version.

**Comment by andreaskatis** [comment]
Hello @SoftPro,

The release comes with prebuilt binaries for external tools such as LustreC. The 'OSX' folder is not relevant to Linux installations. Instead, you should have prebuilt binaries under the 'linux' folder (it is at the same level as the 'OSX' folder). 

In any case, I cannot say for sure if the matlab version is the main culprit. I can say for sure that 2020b works, because this is the latest version I am using. You can try installing that and see if it works.

Finally, since this is an issue with CoCoSim, and not FRET, I would suggest that we continue the discussion over at the CoCoSim Issues page:

https://github.com/NASA-SW-VnV/CoCoSim/issues

If possible, please create a new issue there so that we have all the relevant information within the CoCoSim repository.

**Comment by anmavrid** [comment]
I am closing this issue since it is on the CoCoSim side and the discussion has been moved over to the CoCoSim repository.
